src/enemy/gleerok.c

```diff
@@ -0,0 +1,1258 @@
+            } else {
+                this->unk_82--;
+            }
+        } else {
+            if (this->unk_84->ent2->field_0xf == 0) {
+                super->child = CreateProjectileWithParent(super, GLEEROK_PROJECTILE, 0);
+
+                if (super->child) {
+                    super->child->direction = this->unk_84->filler[0x15];
+                    super->child->type2 = this->unk_84->ent2->frame & 0xf;
+                    super->child->parent = this->unk_84->ent2;
+                    super->child->child = this->unk_84->entities[0];
+                }
+
+                this->unk_84->ent2->actionDelay = 0;
+                if (this->unk_74 == 0) {
+                    this->unk_74 = 1;
+                } else {
+                    super->subAction = 0;
+                    sub_0802EB9C(this);
+                }
+            }
+        }
+    }
+    sub_0802E518(this);
+}
+
+void sub_0802DDD8(GleerokEntity* this) {
+    u32 r2, r8, r4;
+    u32 rv;
+    u32 unk_76 = this->unk_76;
+    u32 uvar1 = 4;
+    if (unk_76) {
+        uvar1 = 6;
+        if (this->unk_79 == 2) {
+            if ((Random() & 1) == 0) {
+                uvar1 = 0xc;
+            }
+        }
+    }
+
+    switch (super->type2) {
+        case 0:
+            rv = sub_0802EA88(this->unk_84);
+            if (rv != 0)
+                break;
+
+            super->type2 = 1;
+            super->field_0xf = 0;
+
+            if ((Random() & 1) == 0) {
+                super->direction = 0;
+                this->unk_7a = (this->unk_84->filler[1] + uvar1) & 0x1f;
+            } else {
+                super->direction = 1;
+                this->unk_7a = (this->unk_84->filler[1] - uvar1) & 0x1f;
+            }
+
+            break;
+        case 1:
+        case 2:
+            if (super->type2 == 1)
+                goto code13;
+            if (this->unk_76 == 0) {
+                r2 = 1;
+                r8 = 2;
+                r4 = 1;
+            } else {
+                r2 = 2;
+                r8 = 6;
+                r4 = 0;
+            }
+
+            if (this->unk_84->ent2->field_0xf == 1) {
+                super->child = CreateProjectileWithParent(super, GLEEROK_PROJECTILE, r2);
+                if (super->child) {
+                    super->child->direction = this->unk_84->filler[0x15];
+                    super->child->type2 = this->unk_84->ent2->frame & 0xf;
+                    super->child->parent = this->unk_84->ent2;
+                    super->child->child = this->unk_84->entities[0];
+                }
+            }
+
+            if (this->unk_74 == 0) {
+                this->unk_74 = r4;
+                this->unk_84->entities[super->field_0xf]->field_0xf = r8;
+
+                if (++super->field_0xf > 5) {
+                    super->field_0xf = 0;
+                }
+            } else {
+                this->unk_74--;
+            }
+
+            if (super->type2 == 1) {
+            code13:
```

Comment:
```suggestion
```
unused

Reply:
It is used (it wouldn't compile otherwise - there's a warning on that).